webcompat/static/js/lib/issue-wizard-popup.js

```diff
@@ -3,39 +3,58 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import $ from "jquery";
+import Mousetrap from "Mousetrap";
 import { sendAnalyticsEvent } from "./wizard/analytics.js";
 
 function Popup() {
   this.init = function () {
+    this.closeButtons = document.querySelectorAll(".popup-modal__close");
+    this.modalTriggers = document.querySelectorAll(".popup-trigger");
+    this.overlay = document.querySelector(".popup-overlay");
+
     this.setUpEvents();
   };
 
   this.setUpEvents = function () {
-    this.popupHandler();
+    this.modalTriggers.forEach(
+      function (trigger) {
+        trigger.addEventListener("click", this.openModal.bind(this));
+      }.bind(this)
+    );
+
+    this.closeButtons.forEach(
+      function (btn) {
+        btn.addEventListener("click", this.closeModal.bind(this));
+      }.bind(this)
+    );
+
+    Mousetrap.bind("esc", _.bind(this.closeModal, this));
+  };
+
+  this.openModal = function (e) {
+    e.preventDefault();
+
+    var popupTrigger = e.target.dataset.popupTrigger;
+    var popupModal = document.querySelector(
+      '[data-popup-modal="'.concat(popupTrigger, '"]')
+    );
+    sendAnalyticsEvent(
+      // transform trigger into camelCase and send as our event
+      popupTrigger.replace(/-([a-z])/g, function (match) {
+        return match[1].toUpperCase();
+      }),
+      "click"
+    );
+    popupModal.classList.add("is--visible");
+    this.overlay.classList.add("is-blacked-out");
   };
 
-  this.popupHandler = function () {
-    var modalTriggers = document.querySelectorAll(".popup-trigger");
-    var overlay = document.querySelector(".popup-overlay");
-    var closeButtons = document.querySelectorAll(".popup-modal__close");
-    modalTriggers.forEach(function (trigger) {
-      trigger.addEventListener("click", function (e) {
-        e.preventDefault();
-        sendAnalyticsEvent("whatIsCompat", "click");
```

Comment:
It turns out we sent this event when opening the "learn more" modal as well. 🤷

Reply:
oops good catch!